### ISSUE TITLE
Add stations/hosts and access points/switches based on their class instead of their name

### ISSUE THREAD

**Issue by MatthiasEckhart** [intrig-unicamp/mininet-wifi#168]
Hello,
why are stations and access points only added if they match the substring `sta` and `ap`, respectively, and not by using class info? Am I missing something that would justify to rely on the node's name? The name of the node could still be used as a fallback, though.

For example:
```
for hostName in topo.hosts():
	cls = topo.nodeInfo(hostName).get('cls')
	if cls is not None:
		if cls == Station or issubclass(cls, Station):
			self.addStation(hostName, **topo.nodeInfo(hostName))
		else:
			self.addHost(hostName, **topo.nodeInfo(hostName))
	else:
		if 'sta' in str(hostName):
			self.addStation(hostName, **topo.nodeInfo(hostName))
		else:
			self.addHost(hostName, **topo.nodeInfo(hostName))
info(hostName + ' ')
```

**Comment by ramonfontes** [comment]
Hello,

Thank you for pointing me that and for your suggestion. I really don't know (and don't remember) why I did that. :) 

Let me propose another suggestion. What do you think?

```for hostName in topo.hosts():                  
            if 'wifi' in str(self.topo):  
                self.addStation(hostName, **topo.nodeInfo(hostName))  
            else:  
                self.addHost(hostName, **topo.nodeInfo(hostName))  
            info(hostName + ' ')```

**Comment by MatthiasEckhart** [comment]
Hello,
thank you for the fast response. I am using mininet-wifi as part of a research project in which I want to replicate a real network setup; thus, in my case the nodes will be named according to the network plan at hand. Replacing `sta` with `wifi` in the if statement would not make anything better and why to check if its in `str(self.topo)` at all?

**Comment by ramonfontes** [comment]
`self.topo` has basically two values:

`mininet.wifi.topo.*` or `mininet.topo.*`. 

The value depends on the topology you have:   
wireless: `sudo mn --wifi --topo *` or   
wired: `sudo mn --topo *`.

**Comment by MatthiasEckhart** [comment]
Correct me, if I am wrong, but this would lead to two issues. First, users would no longer be able to have a mixed topology (consisting of wireless and wired nodes). Second, users may not run Mininet with any of these arguments (`--wifi --topo` or `--topo`), but rather build the topology after Mininet has been started (my case).

**Comment by ramonfontes** [comment]
You still can have a mixed topology, because you do this with scripts and not with `sudo mn`, right? Are you using `topo` when you call Mininet_wifi()?

**Comment by MatthiasEckhart** [comment]
> Are you using topo when you call Mininet_wifi()?

No, I inherit from `Mininet_wifi` and do not pass a topology instance in the constructor. Instead, the topology will be set if users run a certain command via the CLI. Anyway, I forked your repository and adapted the aforementioned parts to my needs. I was just wondering why this has been implemented that way and if there are any implications of using class info whether to add a station or a wired host.

Thank you for the insights. Your suggestion will probably be the best case for most users.
